src/lib/utils.test.ts

```diff
@@ -0,0 +1,102 @@
+import assert from 'assert';
+import { ERC725JSONSchema } from '../types/ERC725JSONSchema';
+import { flattenEncodedData, encodeData } from './utils';
+
+const schema: ERC725JSONSchema[] = [
+  {
+    name: 'SupportedStandards:ERC725Account',
+    key: '0xeafec4d89fa9619884b6b89135626455000000000000000000000000afdeb5d6',
+    keyType: 'Mapping',
+    valueContent: '0xafdeb5d6',
+    valueType: 'bytes',
+  },
+  {
+    name: 'LSP3Profile',
+    key: '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+    keyType: 'Singleton',
+    valueContent: 'JSONURL',
+    valueType: 'bytes',
+  },
+  {
+    name: 'LSP1UniversalReceiverDelegate',
+    key: '0x0cfc51aec37c55a4d0b1a65c6255c4bf2fbdf6277f3cc0730c45b828b6db8b47',
+    keyType: 'Singleton',
+    valueContent: 'Address',
+    valueType: 'address',
+  },
+  {
+    name: 'LSP3IssuedAssets[]',
+    key: '0x3a47ab5bd3a594c3a8995f8fa58d0876c96819ca4516bd76100c92462f2f9dc0',
+    keyType: 'Array',
+    valueContent: 'Number',
+    valueType: 'uint256',
+    elementValueContent: 'Address',
+    elementValueType: 'address',
+  },
+];
+
+describe('#flattenEncodedData', () => {
+  it('should flatten encodedData (with one key)', () => {
+    const encodedData = {
+      LSP3Profile: {
+        value:
+          '0x6f357c6a2404a2866f05e53e141eb61382a045e53c2fc54831daca9d9e1e039a11f739e1696670733a2f2f516d5154716865424c5a466e5155787535524473387441394a746b78665a714d42636d47643973756b587877526d',
+        key: '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      },
+    };
+
+    assert.deepStrictEqual(flattenEncodedData(encodedData), [
+      {
+        value:
+          '0x6f357c6a2404a2866f05e53e141eb61382a045e53c2fc54831daca9d9e1e039a11f739e1696670733a2f2f516d5154716865424c5a466e5155787535524473387441394a746b78665a714d42636d47643973756b587877526d',
+        key: '0x5ef83ad9559033e6e941db7d7c495acdce616347d28e90c7ce47cbfcfcad3bc5',
+      },
+    ]);
+  });
```

Comment:
I don't like when duplicating the values in a test case, because then if you update the `key` in encoded data, you will also have to update it in the assert.

What about:

```js
assert.deepStrictEqual(flattenEncodedData(encodedData), [
      {
        value: encodedData.LSP3Profile.value,
        key: encodedData.LSP3Profile.key,
      },
    ]);
  });
```

then it becomes much readable and it is easier to understand what this function is supposed to do + no duplicate